packages/smooth_app/lib/pages/product/nutrition_page_loaded.dart

```diff
@@ -360,28 +361,40 @@ class _NutritionPageLoadedState extends State<NutritionPageLoaded> {
         label: Text(appLocalizations.nutrition_page_add_nutrient),
       );
 
-  Future<bool> _showCancelPopup(AppLocalizations localizations) async =>
-      await showDialog<bool>(
-        context: context,
-        builder: (BuildContext context) => AlertDialog(
-          shape: const RoundedRectangleBorder(
-            borderRadius: ROUNDED_BORDER_RADIUS,
-          ),
-          title: Text(localizations.general_confirmation),
-          content: Text(localizations.nutrition_page_close_confirmation),
-          actions: <TextButton>[
-            TextButton(
-              child: Text(localizations.cancel.toUpperCase()),
-              onPressed: () => Navigator.pop(context, false),
-            ),
-            TextButton(
-              child: Text(localizations.close.toUpperCase()),
-              onPressed: () => Navigator.pop(context, true),
+  Future<bool> _showCancelPopup(AppLocalizations localizations) async {
```

Comment:
@AshAman999 I think I understand now. I suggest these changes:
```dart
/// Returns true if the page was not edited or if the user confirms the changes.
///
/// The returned value will be used by onWillPop.
Future<bool> _checkConfirmedChanges(AppLocalizations localizations) async {
```
and `localizations.okay` instead of `localizations.close`.

Reply:
did the changes,